packages/main/src/SliderBase.js

```diff
@@ -0,0 +1,648 @@
+			touchstart: "touchmove",
+		};
+	}
+
+	get labelItems() {
+		return this._labelItems;
+	}
+
+	get classes() {
+		return {
+			labelContainer: {
+				"ui5-slider-hidden-labels": this._labelsOverlapping,
+			},
+		};
+	}
+
+	onEnterDOM() {
+		ResizeHandler.register(this, this._resizeHandler);
+		this.addEventListener("mouseover", this._mouseOverHandler);
+		this.addEventListener("mouseout", this._mouseOutHandler);
+	}
+
+	onExitDOM() {
+		ResizeHandler.deregister(this, this._handleResize);
+		this.removeEventListener("mouseover", this._mouseOverHandler);
+		this.removeEventListener("mouseout", this._mouseOutHandler);
+		this._labelItems = null;
+	}
+
+	onAfterRendering() {
+		// Only call if the resize is triggered by a state changes other than
+		// the ones that occured on the previous resize and those caused by user interaction.
+		if (this.notResized) {
+			this._resizeHandler();
+		}
+	}
+
+	_ontouchstart(event) {
+		this._onmousedown(event);
+	}
+
+	_ontpointerdown(event) {
+		this._onmousedown(event);
+	}
+
+	/** Shows the tooltip(s) if the <code>showTooltip</code> property is set to true
+	 *
+	 * @private
+	 */
+	_onmouseover(event) {
+		if (this.showTooltip) {
+			this._tooltipVisibility = "visible";
+		}
+	}
+
+	/**
+	 * Hides the tooltip(s) if the <code>showTooltip</code> property is set to true
+	 *
+	 * @private
+	 */
+	_onmouseout(event) {
+		if (this.showTooltip) {
+			this._tooltipVisibility = "hidden";
+		}
+	}
+
+	/**
+	 * Handle the responsiveness of the Slider's UI elements when resing
+	 *
+	 * @private
+	 */
+	_handleResize() {
+		if (!this.showTickmarks) {
+			return;
+		}
+
+		// Mark resizing to avoid unneccessary calls to that function after rendering
+		this.notResized = false;
+
+		// Convert the string represented calculation expression to a normal one
+		// Check the distance  in pixels exist between every tickmark
+		const spaceBetweenTickmarks = this._spaceBetweenTickmarks();
+
+		// If the pixels between the tickmarks are less than 8 only the first and the last one should be visible
+		// In such case the labels must correspond to the tickmarks, only the first and the last one should exist.
+		if (spaceBetweenTickmarks < 8) {
+			this._tickmarksBackground = `linear-gradient(to right, currentColor 1px, transparent 0) 0 center / calc(100% - 1px) 100% repeat-x`;
+			this._hiddenTickmarks = true;
+			this._labelsOverlapping = true;
+		} else {
+			this._drawDefaultTickmarks(this.step, this.max, this.min);
+			this._hiddenTickmarks = false;
+		}
+
+		if (this.labelInterval <= 0 || this._hiddenTickmarks) {
+			return;
+		}
+
+		// Cache the labels if not yet fetched
+		if (!this._labels) {
+			this._labels = this.shadowRoot.querySelectorAll(".ui5-slider-labels li");
+		}
+
+		// Check if there are any overlapping labels.
+		// If so - only the first and the last one should be visible
+		this._labelsOverlapping = Array.prototype.some.call(this._labels, label => label.scrollWidth > label.clientWidth);
```

Comment:
`[...this._labels].some` is the preferred way to convert a DOMNodeList to an array.

Reply:
This is cool :)